contexts/AuthContext.tsx

```diff
@@ -0,0 +1,139 @@
+"use client";
+
+import React, { createContext, useContext, useEffect, useState } from "react";
+import { AuthService, AuthUser } from "../lib/appwrite/auth";
+
+interface AuthContextType {
+  user: AuthUser | null;
+  isLoading: boolean;
+  isAuthenticated: boolean;
+  signInWithEmail: (
+    email: string,
+  ) => Promise<{ success: boolean; error?: string; userId?: string }>;
+  verifyEmailOTP: (
+    userId: string,
+    otp: string,
+  ) => Promise<{ success: boolean; error?: string }>;
+  signInWithGoogle: () => Promise<{ success: boolean; error?: string }>;
+  signInWithApple: () => Promise<{ success: boolean; error?: string }>;
+  signOut: () => Promise<void>;
+  refreshUser: () => Promise<void>;
+}
+
+const AuthContext = createContext<AuthContextType | undefined>(undefined);
+
+export const useAuth = () => {
+  const context = useContext(AuthContext);
+  if (context === undefined) {
+    throw new Error("useAuth must be used within an AuthProvider");
+  }
+  return context;
+};
+
+interface AuthProviderProps {
+  children: React.ReactNode;
+}
+
+export const AuthProvider: React.FC<AuthProviderProps> = ({ children }) => {
+  const [user, setUser] = useState<AuthUser | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isAuthenticated, setIsAuthenticated] = useState(false);
+
+  const refreshUser = async () => {
+    try {
+      const result = await AuthService.getCurrentUser();
+      if (result.user) {
+        setUser(result.user);
+        setIsAuthenticated(true);
+      } else {
+        setUser(null);
+        setIsAuthenticated(false);
+      }
+    } catch (error) {
+      setUser(null);
+      setIsAuthenticated(false);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const signInWithEmail = async (email: string) => {
```

Comment:
do we need signInWithEmail is we're not using passwords?

Reply:
`signInWithEmail ` renamed to `sendEmailOTP` in https://github.com/Ditectrev/Practice-Exams-Platform/pull/70/commits/26fb3665a5751b653395727d5048d86b82855e29 to avoid future confusions.